src/components/thumbnail/documents-type-collection.tsx

```diff
@@ -64,11 +65,14 @@ function getSectionDocs(section: NavTabSectionModelType,
               publishedDocs[source] = doc;
             }
           }
+          const teacher = classStore.getUserById(publishedDocs[source].uid)?.type === "teacher";
+          if (teacher) {
+            publishedDocs[source].setProperty("isTeacherDocument", "true");
+          }
```

Comment:
I think we're going to have to do this differently. One of two things is true: either
1. we want to explicitly store `isTeacherDocument` as a property on the document, in which case it should presumably be set when the document is created; or
2. we want to treat `isTeacherDocument` as a pseudo-property like `starred`, which is never stored with the document but the matching code pretends that it is.

By setting the `isTeacherDocument` property here, we are potentially causing it to be written/synchronized to firebase whenever it is viewed in the NavTabs. I think there's a discussion to be had here about the best way to proceed.

Having stared at this some more, I think the best way to handle the `isTeacherDocument` pseudo-property, is to:
1. add an `isTeacher(uid: string)` method to the `ClassModel`
2. add an `options` argument to `DocumentModel.matchProperties` which takes an `isTeacherDocument(document: DocumentModelType)` callback so that `matchProperties` can determine whether or not to match the `isTeacherDocument` pseudo-property based on the result of the callback without adding a direct dependency on the stores/ClassModel.
3. have clients that call `matchProperties` pass in the `isTeacherDocument` callback if they (could) care about the distinction.

That way we don't have to store the property with the document redundantly, and we don't introduce any dependencies that aren't already present.